docs/developers/architecture.md

```diff
@@ -0,0 +1,147 @@
+# Architectural Design
+
+## Parsing XML Files and Transformations for `xmlupload` and `validate-data`
+
+### Overview
+
+```mermaid
+---
+title: Overview of Code Flow for xmlupload and validate-data
+---
+stateDiagram-v2
+
+state "etree Root" as eroot1
+state "etree Root" as eroot2
+state "etree Root" as eroot3
+state "Root Modifications" as rootwork
+state "etree to Python Representation" as pywork
+state "CLI: xmlupload" as transform
+state "ParsedResource" as parsedres1
+state "ParsedResource" as parsedres2
+state "ParsedResource" as parsedres3
+state "IntermediaryResource" as intermediaryres
+state "CLI: validate-data" as valdata
+
+[*]-->rootwork
+state rootwork {
+    eroot1-->eroot2: xsd validation success
+    eroot1-->[*]: xsd validation error raised
+}
+eroot2-->pywork
+state pywork {
+    eroot3-->parsedres1: transform representation
+}
+pywork-->transform
+state transform {
+    ResourceInputConversionFailure-->[*]: transformation error raised
+    parsedres2-->intermediaryres: transformation success
+    parsedres2-->ResourceInputConversionFailure: transformation failure
+}
+pywork-->valdata
+state valdata {
+    parsedres3-->ResourceDeserialised: transformations
+}
+```
+
+### Parsing XML Files
+
+<!-- markdownlint-disable MD013 -->
+
+```mermaid
+---
+title: Parsing of XML File and Transformations Into ParsedResource
+---
+stateDiagram-v2
+state "Transform etree Root" as eroot
+state "Transform etree Root into Python Representation" as transpy
+state "Resource etree" as resetree1
+state "Resource etree" as resetree2
+state "ParsedResource" as respars
+state "Value etree" as valtree
+state "IIIF/bitstream etree" as filetree
+state "ParsedValue" as valpars
+state "ParsedFileValue" as filepars
+state "Transform Resource" as transres
+state "Transform Value" as transval
+state "Transform FileValue" as transfile
+state "Transformed Content" as rescontent
+
+[*]-->r1: Parse file
+
+r4-->transpy
+    state eroot {
+        state "etree root" as r1
+        state "etree root" as r2
+        state "etree root" as r3
+        state "etree root" as r4
+        r1-->r2: Remove Comments
+        r2-->[*]: xsd validation failure
+        r2-->r3: xsd validation success
+        r3-->r4: make localnames
+    }
+    state transpy {
+        resetree1-->transres
+        resetree1-->transval: extract values
+        resetree1-->transfile: extract file or iiif
+        
+        state transres {
+            resetree2-->rescontent: transform special tags</br></br>resolve prefixes into absolute IRIs
```

Comment:
The parser seems to be forgiving, but AFAIK, `</br>` is not correct HTML. `<br/>` seems more correct to me.